lit_gpt/lora.py

```diff
@@ -390,11 +456,51 @@ class Config(BaseConfig):
             "This scaling helps to reduce the need to retune hyperparameters when we vary r"
             https://arxiv.org/pdf/2106.09685.pdf (section 4.1)
         dropout: dropout that is applied on the input in the LoRA branch (before multiplying by matrix A)
+        to_*: either apply LoRA to the specified weights or not
     """
 
     r: int = 0.0
     alpha: int = 1.0
     dropout: float = 0.0
+    to_query: bool = False
+    to_key: bool = False
+    to_value: bool = False
+    to_projection: bool = False
+    to_mlp: bool = False
+    to_head: bool = False
+
+    @property
+    def mlp_class(self) -> Type:
+        # `self._mlp_class` cannot be the type to keep the config json serializable
+        obj = lit_gpt.lora if self.to_mlp else lit_gpt.model
+        return getattr(obj, self._mlp_class)
+
+class GptNeoxMLP(nn.Module):
```

Comment:
This class (and LLaMAMLP) could subclass the base classes so that forward doesn't need to be implemented. `GPT` does it already

Reply:
Good idea.
Done.
The only thing is that (as I can see) it's not possible to name parent class as `BaseMLP` since we have two MLP classes to choose from. So it brakes naming a lil bit.